source/features/conversation-activity-filter.tsx

```diff
@@ -147,17 +146,22 @@ function addWidget(position: Element): void {
 	);
 }
 
-function init(): void {
-	const observer = observe(':is(.gh-header-sticky .meta, .gh-header-meta .flex-auto):not(.rgh-conversation-activity-filter)', {
-		add: addWidget
-	});
-	deinit.push(observer.abort);
+async function init(): Promise<void> {
+	const headerSeectors = ':is(.gh-header-sticky .meta, .gh-header-meta .flex-auto):not(.rgh-conversation-activity-filter)';
+	await elementReady(headerSeectors);
+	const headers = select.all(headerSeectors);
```

Comment:
These 2 aren't compatible. `element-ready` resolves on the first one, so that means `.all` won't match the second one on some page loads

Reply:
> These 2 aren't compatible. `element-ready` resolves on the first one, so that means `.all` won't match the second one on some page loads.

It works since I am not doing `waitForChildren:false`

Reply:
I think you're confused about how that works. `waitForChildren` has no effect on this. If anything, it's worse because `false` means it resolves as soon as the opening tag is found.

Reply:
Given:

```html
<mango>
	<banana/>
</mango>
<pear></pear>
<mango></mango>
```

1. `elementReady('mango')` will resolve when line 4 is parsed. `select.all('mango')` will not include the second `<mango>`
1. `elementReady('mango', {waitForChildren:false})` will resolve when line 1 is parsed. `select.all('mango')` will definitely not include the second `<mango>`

Reply:
So I should just wait for the sticky, that is second?

Reply:
If you know they’re exactly too, you can just use 2 elementReady, one customized for each